src/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/AngularVelocity.cpp

```diff
@@ -0,0 +1,63 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/AngularVelocity.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+namespace state
+{
+namespace coordinatessubsets
+{
+
+using ostk::math::object::Vector3d;
+
+using ostk::physics::coord::Position;
+using ostk::physics::coord::Velocity;
+
+AngularVelocity::AngularVelocity(const Shared<const AttitudeQuaternion>& aAttitudeQuaternionSPtr, const String& aName)
+    : CoordinatesSubset(aName, 3),
+      attitudeQuaternionSPtr_(aAttitudeQuaternionSPtr)
+{
+}
+
+AngularVelocity::~AngularVelocity() {}
+
+VectorXd AngularVelocity::inFrame(
+    const Instant& anInstant,
+    const VectorXd& aFullCoordinatesVector,
+    const Shared<const Frame>& fromFrame,
+    const Shared<const Frame>& toFrame,
+    const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
+) const
+{
+    const VectorXd coordinates = aCoordinatesBrokerSPtr->extractCoordinate(aFullCoordinatesVector, *this);
+
+    const VectorXd attitudeCoordinatesInFrame = this->attitudeQuaternionSPtr_->inFrame(
+        anInstant, aFullCoordinatesVector, fromFrame, toFrame, aCoordinatesBrokerSPtr
+    );
+    const Quaternion quaternionInFrame = AttitudeQuaternion::coordinatesToQuaternion(attitudeCoordinatesInFrame);
```

Comment:
same, use built-in methods?

Reply:
try protected/friend visiblity

Reply:
Ok so I have tried. The only way that one can ensure that only some classes in astro can access it is the combination of the `friend` modifier and forward declaration. I have tried several times, but I don't know if we can actually get away with it. I've "exhausted" stackoverflow+chat gpt. It's quite a niche case: static private, be-friending 2 classes, which are not on the same level and which depend on the AttitudeQuaternion file itself.

If we are going to move this to the attitude object in the near future is it ok to leave it as is?

cc @Derollez

Reply:
Yup that's good on my side ✅

Reply:
(resolving)